Mu2eG4/src/constructDiskCalorimeter.cc

```diff
@@ -289,48 +288,117 @@ namespace mu2e {
-
-           pv = new G4PVPlacement(rotPipe1,G4ThreeVector(0,0,z), pipe1Log, "caloPipePV", frontPanelFoamLog, false, ipipe, false);
+           //G4Torus* pipe1 = new G4Torus("caloPipe",pipeRadius-pipeThickness, pipeRadius, pipeTorRadius[ipipe],angle, CLHEP::pi-2*angle);
+           // reduce the angle of large bending to 99.0% of the original, small bending 99.5% of the original 
+           G4Torus* pipe1 = new G4Torus("caloPipe1", pipeRadius-pipeThickness, pipeRadius, pipeTorRadius[ipipe], -angle*0.9975, 2*angle*0.9975);
+           G4Torus* pipe2 = new G4Torus("caloPipe2", pipeRadius-pipeThickness, pipeRadius, smallRadius, angle-sAngle+CLHEP::pi, sAngle);
+           //G4Tubs*  pipe2 = new G4Tubs("caloPipe2",pipeRadius-pipeThickness, pipeRadius,0.5*length,0,CLHEP::twopi);       
+           G4LogicalVolume* pipe1Log = caloBuildLogical(pipe1, pipeMaterial, "caloPipe1Log",isPipeVisible,G4Color::Cyan(),isPipeSolid,forceEdge);
+           G4LogicalVolume* pipe2Log = caloBuildLogical(pipe2, pipeMaterial, "caloPipe2Log",isPipeVisible,G4Color::Green(),isPipeSolid,forceEdge);
+           //G4LogicalVolume* pipe2Log = caloBuildLogical(pipe2, pipeMaterial, "caloPipe2Log",isPipeVisible,G4Color::Cyan(),isPipeSolid,forceEdge);
+
+           // calculate the parameters of the strait pipes
+           // minus 2.0 mm from the manifold pipe's most inner radius to avoid the overlap.
+           double xEnd = (FPCoolPipeTorRadius -FPCoolPipeRadius) * sin(straitEndPhi[ipipe] * CLHEP::pi / 180.);
+           double yEnd = (FPCoolPipeTorRadius -FPCoolPipeRadius) * cos(straitEndPhi[ipipe] * CLHEP::pi / 180.);
+           double xStart = sxPos - smallRadius * cos(straitEndPhi[ipipe] * CLHEP::pi / 180.);
+           double yStart = syPos + smallRadius * sin(straitEndPhi[ipipe] * CLHEP::pi / 180.);
+           double sLength = sqrt((xEnd - xStart)*(xEnd - xStart) + (yEnd - yStart)*(yEnd - yStart));
+           double zRotateAngle = std::atan((yEnd - yStart) / (xEnd - xStart));
+
+           // rotation coordinate
+           //G4RotationMatrix* sPipeRotate = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
+           G4RotationMatrix* sPipeRotate1 = new G4RotationMatrix();
+           sPipeRotate1 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate1 -> rotateY((-0.5*CLHEP::pi + zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate2 = new G4RotationMatrix();
+           sPipeRotate2 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate2 -> rotateY((-0.5*CLHEP::pi - zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate3 = new G4RotationMatrix();
+           sPipeRotate3 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate3 -> rotateY((-1.5*CLHEP::pi + zRotateAngle) * CLHEP::rad);
+           G4RotationMatrix* sPipeRotate4 = new G4RotationMatrix();
+           sPipeRotate4 -> rotateX(0.5 * CLHEP::pi);
+           sPipeRotate4 -> rotateY((-1.5*CLHEP::pi - zRotateAngle) * CLHEP::rad);
+           //reg.add(sPipeRotate);
```

Comment:
As of now, G4RotationMatrix* sPipeRotateN = new G4RotationMatrix(); are memory leaks. Replace with
G4RotationMatrix* sPipeRotate1 = reg.add(new G4RotationMatrix());

Reply:
Done

Reply:
I do not see the use of reg.add(new G4RotationMatrix()) in the lines where new G4RotationMatrix() is used,
so those are still memory leaks